tensorflow/java/src/main/java/org/tensorflow/Tensor.java

```diff
@@ -244,9 +237,10 @@
    * @throws IllegalArgumentException If the tensor datatype or shape is not compatible with the
    *     buffer
    */
-  @SuppressWarnings("unchecked")
   public static <T> Tensor<T> create(Class<T> type, long[] shape, ByteBuffer data) {
-    return (Tensor<T>) create(DataType.fromClass(type), shape, data);
+    @SuppressWarnings("unchecked")
```

Comment:
Local variable ret must be introduced to allow @SuppressWarnings to be moved inside the method defn.